src/path/path.rs

```diff
@@ -0,0 +1,105 @@
+use std::ffi::OsStr;
+
+use crate::path::PathBuf;
+use crate::{fs, io};
+
+/// This struct is an async version of [`std::path::Path`].
+///
+/// [`std::path::Path`]: https://doc.rust-lang.org/std/path/struct.Path.html
+pub struct Path {
+    inner: OsStr,
```

Comment:
I'm wondering whether it might perhaps be easier to store a `std::path::PathBuf` internally, and delegate all non-io methods to it?

That way we can prevent having to re-implement some of the trickier logic such as going between `OsStr` and `Path`.

Reply:
How about using std::path::Path in Path, and std::path::PathBuf in PathBuf?

Reply:
@wassasin ah oops yeah; that's what I meant :sweat_smile: